src/node_contextify.h

```diff
@@ -210,6 +211,69 @@ v8::Maybe<bool> StoreCodeCacheResult(
     bool produce_cached_data,
     std::unique_ptr<v8::ScriptCompiler::CachedData> new_cached_data);
 
+class NodeRealm final : public node::MemoryRetainer {
+ public:
+  static v8::Local<v8::FunctionTemplate> GetConstructorTemplate(
+      IsolateData* isolate_data);
+  static void CreatePerIsolateProperties(IsolateData* isolate_data,
+                                         v8::Local<v8::ObjectTemplate> target);
+  static void RegisterExternalReferences(
+      node::ExternalReferenceRegistry* registry);
+
+  NodeRealm(node::Environment* env, v8::Local<v8::Object> obj);
+
+  static void New(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void Start(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void Load(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void SignalStop(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void Stop(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void InternalRequire(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void TryCloseAllHandles(
+      const v8::FunctionCallbackInfo<v8::Value>& args);
+
+  struct NodeRealmScope : public v8::EscapableHandleScope,
+                          public v8::Context::Scope,
+                          public v8::Isolate::SafeForTerminationScope {
+   public:
+    explicit NodeRealmScope(NodeRealm* w);
+    ~NodeRealmScope();
+
+   private:
+    NodeRealm* w_;
```

Comment:
Non-blocking nit: not really a fan of `w_` as a member variable name. Something more descriptive would be ideal. `realm_` perhaps?